### PR TITLE
add https support 

### PR DESCRIPTION
**Pull request by Rajat-Goyal**
Add support for https, #233  
I and @agaurav77 are still working on more suitable tests.

### CODE REVIEW

File: bin/ssh_scan_api [hunk 1]
```diff
@@ -22,6 +22,18 @@ opt_parser = OptionParser.new do |opts|
     exit
   end
 
+  opts.on("--https", "Use HTTPS") do
+    options[:https] = true
+  end
```
Comment (claudijd) [inline bin/ssh_scan_api]: Does this just serve a self-signed cert if the key and crt files aren't provided?

Comment (claudijd) [inline bin/ssh_scan_api]: I tried running the following and it didn't serve a self-signed cert by default....

./bin/ssh_scan_api --https

**Comment by claudijd** [comment]
I'm making some slight changes to this, but will be landing shortly

File: bin/ssh_scan_api (hunk 1 above)
Comment (agaurav77) [inline bin/ssh_scan_api]: https would work like this
./bin/ssh_scan_api --https --https-key <keyfile> --https-crt <crtfile>

We were initially going to go for generation of self signed certs using OpenSSL everytime, but we thought you might want to add in one of LetsEncrypt based certificates on your own. If you want certs to be generated everytime we can add that back.

**Comment by claudijd** [comment]
Changes I made can be found here, basically don't provide an HTTP option at all.

By default the API will be served as self-signed as generated by the library and you can override the cert/key using the options you added to the api command line.  This should make it easy for people to use SSL in the future.

https://github.com/mozilla/ssh_scan/commit/bbbbb6854510f138d65bd0535cab7585186e1c9b

**Pull request merged**